### PR TITLE
Remove *math* parameter of various mathtext internal APIs.

### LINKED ISSUE #21687: [Bug]: math_fontfamily="cm" prevents replacement of minus by unicode minus in `\mathdefault`
### Bug summary

All's in the title (and the examples).

### Code for reproduction

```python
from pylab import *
figtext(.5, .6, r"$\mathdefault{a-b}$", math_fontfamily="dejavusans")
figtext(.5, .5, r"$\mathdefault{a-b}$", math_fontfamily="cm")
```

### Actual outcome

test
Both lines use the text default font of DejaVu Sans, but setting math_fontfamily to cm prevented replacement of minus by unicode minus.

### Expected outcome

Unicode minus in both cases

### Additional information

(I probably have a fix that'll come after tearing down most mathtext deprecations that expire in 3.6, namely getting rid of the `math` kwarg in the various font classes and instead make the minus->unicode_minus replacement earlier, during the parsing stage.)

### Operating system

linux

### Matplotlib Version

3.5

### Matplotlib Backend

qt5agg

### Python version

39

### Jupyter version

ENOSUCHLIB

### Installation

_No response_

### PR DESCRIPTION
**Pull request by anntzer**
The *math* parameter is passed through many layers of the call stack
but is ultimately only used for a single purpose: deciding whether to
replace the ASCII hyphen by a (longer) unicode minus.  Instead of doing
that, just do the substitution at the parsing stage.  In particular,
this fixes problematic unicode minus support with the "cm" fontset.
    
This patch also reverts a significant part of 52003e4, as LogFormatters
no longer need to pass unicode minuses in mathtext -- everything gets
converted by mathtext.  Likewise, this change also invalidates the
test_log_scales baseline image (old, buggy wrt. unicode minus); replace
it by a test that the drawn ticks are as expected (which was the intent
in 90c1aa3).

Closes #21687.

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by oscargus** [comment]
How does this affect the possibility to support `\text`?

**Comment by anntzer** [comment]
I think it's OK?  Your implementation at #22173 should not be affected too much by this, as long as it doesn't perform the same dash-to-unicodeminus replacement as is now done by symbol().

**Comment by oscargus** [comment]
OK! Just was a bit worried that the long dash was already in the string, so that $\text{foo-bar}$ ended up as $\text{foo−bar}$, so that it had to be reverted in case of \text.

Not sure I will finish #22173 (especially since I read that you have one), but we'll see.

**Comment by anntzer** [comment]
I don't have an implementation of `\text`?  I think #22173 is probably the closest there is.

Replacment to unicode minus occurs within `symbol`, which `text` doesn't hit, I think?

**Comment by oscargus** [comment]
OK! Just have some memory of reading that you had a branch hanging around (but I may be wrong and either I didn't read it or it was someone else).

Ahh, I see now. Primarily read the description, which caused my doubt.

**Comment by anntzer** [comment]
I think I gave it a try at some point, but am not sure it's really ready either, so let's try to get yours in.

**Review by oscargus — APPROVED** [review]
Not fully sure why the image tests are removed, but the general change seems to make sense.

**Comment by anntzer** [comment]
The old image had a non-unicode minus (due to #21687); this patch changes the image.  But we try to avoid adding new (or modified) baseline images if possible (due to repo size concerns); here the test was really to check what are the ticks so now I just do that instead.

**Comment by jklymak** [comment]
Not sure if all the failures are related....

**Comment by anntzer** [comment]
Should be good to go now.

File: lib/matplotlib/_mathtext.py [hunk 1]
```diff
@@ -1624,8 +1616,8 @@ class _MathStyle(enum.Enum):
         SCRIPTSTYLE = enum.auto()
         SCRIPTSCRIPTSTYLE = enum.auto()
 
-    _binary_operators = set(r'''
-      + * -
+    _binary_operators = set('''
+      + * - \N{MINUS SIGN}'''r'''
```
Comment (timhoffm) [inline lib/matplotlib/_mathtext.py:1620]: Simpler formatting:
```suggestion
    _binary_operators = set(
      '+ * - \N{MINUS SIGN}'
      r'''
```

**Review by timhoffm — APPROVED** [review]
May self-merge after addressing the formatting suggestion.

File: lib/matplotlib/_mathtext.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/_mathtext.py:1620]: done

**Pull request merged**